static/src/javascripts/projects/commercial/initialConsentState.ts

```diff
@@ -0,0 +1,38 @@
+import { onConsentChange } from '@guardian/consent-management-platform';
+import type { ConsentState } from '@guardian/consent-management-platform/dist/types';
+
+/**
+ * return a promise containing the first consent state provided by the user
+ * as soon as it becomes available.
+ * NOTE: depending on where this function is eventually used, it might be more appropriate
+ * for it to be defined in the consent-management-platform
```

Comment:
Subjective and up to you but Is it worth adding something along the lines of... this will only only resolve consent state once whereas CMP `onConsentChange` will notify on all consent state changes.